Source/WebCore/loader/cache/CachedResource.cpp

```diff
@@ -184,7 +184,17 @@ void CachedResource::load(CachedResourceLoader& cachedResourceLoader)
     // We query the top document because new frames may be created in pagehide event handlers
     // and their backForwardCacheState will not reflect the fact that they are about to enter page
     // cache.
-    if (auto* topDocument = frame.mainFrame().document()) {
+    if (!dynamicDowncast<LocalFrame>(frame.mainFrame())) {
+        // FIXME: Fix with website isolation work
+        failBeforeStarting();
```

Comment:
Let's just skip the use of topDocument if the main frame isn't local, instead of failing all loads or early returning.